// @aws-sdk/client-polly/dist-es/commands/GetSpeechSynthesisTaskCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetSpeechSynthesisTaskInputFilterSensitiveLog, GetSpeechSynthesisTaskOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetSpeechSynthesisTaskCommand, serializeAws_restJson1GetSpeechSynthesisTaskCommand, } from "../protocols/Aws_restJson1";
export class GetSpeechSynthesisTaskCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "PollyClient";
        const commandName = "GetSpeechSynthesisTaskCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetSpeechSynthesisTaskInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetSpeechSynthesisTaskOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restJson1GetSpeechSynthesisTaskCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restJson1GetSpeechSynthesisTaskCommand(output, context);
    }
}
